### PR TITLE
Service chart enhancements for portal-app

### PR DESCRIPTION
**Pull request by mbacchi**
Significant updates include:

* Removed 2 if statements from podDisruptionBudget. Now it will check if `.Values.podDisruptionBudget` is set. If true, we will use that value, otherwise we will provide a default of `maxUnavailable: 25%`.
* Removed podDisruptionBudget default from values.yaml. This maxUnavailable default interfered with any requirement to use minAvailable in subcharts. Now the default is defined in the pdb.yaml, documented in the above bullet point.
* Allow podDisruptionBudget selector labels to be overridden.
* Added jobs.yaml to enable jobs that aren't considered "pre-release" jobs.

Resolves: https://codecademy.atlassian.net/browse/DEVOPS-6440

This has been tested (using `make k8s-template` or the equivalent command) against the following:
  * workspace-service
  * portal-app in monorepo
  * codecademy PR env
  * codecademy zoo env
  * codecademy staging

### CODE REVIEW

File: charts/service/values.yaml [hunk 1]
```diff
@@ -102,7 +102,7 @@ datadog: {}
   # service: service_name
   # version: version
 
-preReleaseJob:
+preReleaseJobs:
```
Comment (leonpham0) [inline charts/service/values.yaml:105]: I think this should stay as `Job`, our template uses `preReleaseJob` not `Jobs` (`.Values.preReleaseJob.enabled`).

File: charts/service/Chart.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: service
 description: A generic k8s service chart
 type: application
-version: 1.7.5
+version: 1.7.7
```
Comment (leonpham0) [inline charts/service/Chart.yaml:5]: Why the jump to 7 and skipping 6?

Comment (mbacchi) [reply charts/service/Chart.yaml:5]: This earlier [PR](https://github.com/codecademy-engineering/helm-charts/pull/96) has 1.7.6.

File: charts/service/values.yaml (hunk 1 above)
Comment (mbacchi) [reply charts/service/values.yaml:105]: Yep, this is accurate, no other repos have the string `preReleaseJobs` yet 23 have the string `preReleaseJob`. Somehow this wasn't working for me earlier. Will fix.

File: charts/service/Chart.yaml (hunk 2 above)
Comment (lindsaylandry) [reply charts/service/Chart.yaml:5]: this is a big change, so maybe bump to `1.8.0`

File: charts/service/templates/jobs.yaml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            {{- with .env }}
+            {{- toYaml . | nindent 12 }}
+            {{- end }}
+            {{- with $datadog }}
+            {{- if and .service .env }}
+            - name: DD_AGENT_HOST
+              valueFrom:
+                fieldRef:
+                  fieldPath: status.hostIP
+            - name: DATADOG__ENABLED
```
Comment (lindsaylandry) [inline charts/service/templates/jobs.yaml:44]: change these to `DD_`

File: charts/service/values.yaml (hunk 1 above)
Comment (mbacchi) [reply charts/service/values.yaml]: Done

File: charts/service/Chart.yaml (hunk 2 above)
Comment (mbacchi) [reply charts/service/Chart.yaml]: Done

**Review by anaharris — APPROVED** [review]

**Pull request merged**